pkg/sql/logictest/testdata/logic_test/enums

```diff
@@ -989,3 +989,37 @@ INSERT INTO enum_referenced VALUES ('howdy')
 
 statement error pq: foreign key violation: "enum_origin" row x='hello' has no match in "enum_referenced"
 ALTER TABLE enum_origin ADD FOREIGN KEY (x) REFERENCES enum_referenced (x)
+
+# Tests for ALTER COLUMN SET DATA TYPE.
```

Comment:
Can you add some tests where
1. The conversion to enum is invalid
2. enum to another enum
3. maybe enum to the same enum to make sure the type equivalency check is okay and we perform a no-op.

Reply:
Done 1, 3. 2 isn't possible directly (pg doesnt allow enum -> enum casts), it needs `USING EXPRESSION`.